doc/developer/design/20231113_optimizer_notice_catalog.md

```diff
@@ -0,0 +1,480 @@
+user-initiated re-creation of the associated catalog entry.
+
+### SOC2 Compliance
+
+Some of the notices print literal constraints appearing in the
+
+```sql
+CREATE MATERIALIZED VIEW AS <query>
+CREATE VIEW AS <query>
+```
+
+statements. Since literal constrains appearing in these statements are
+[classified as "Customer Data"][data_mgmt_policy], we should obfuscate those in
+`redacted_~` columns similarly to the `redacted_create_sql` columns proposed in
+MaterializeInc/materialize#19850.
+
+## Success Criteria
+
+<!--
+What does a solution to this problem need to accomplish in order to
+be successful?
+
+The criteria should help us verify that a proposed solution would solve
+our problem without naming a specific solution. Instead, focus on the
+outcomes we hope result from this work. Feel free to list both qualitative
+and quantitative measurements.
+-->
+
+Stakeholders have agreed on:
+
+1. The primary use cases and UX workflows that the schema will facilitate.
+2. The design of a catalog schema that contains the optimizer notices
+   corresponding to the current catalog state.
+3. The design of code that will maintain the contents of that schema at runtime,
+   including the mechanisms used to align the externally set notice status with
+   the internal system state.
+
+## Out of Scope
+
+<!--
+What does a solution to this problem not need to address in order to be
+successful?
+
+It's important to be clear about what parts of a problem we won't be solving
+and why. This leads to crisper designs, and it aids in focusing the reviewer.
+-->
+
+1. Design details about the actual console UX beyond basic wireframe diagrams.
+2. Implementation details about the integration path for optimizer notices
+   associated with `SELECT` queries (although the design should have at least
+   some idea how these will work out).
+
+## Solution Proposal
+
+<!--
+What is your preferred solution, and why have you chosen it over the
+alternatives? Start this section with a brief, high-level summary.
+
+This is your opportunity to clearly communicate your chosen design. For any
+design document, the appropriate level of technical details depends both on
+the target reviewers and the nature of the design that is being proposed.
+A good rule of thumb is that you should strive for the minimum level of
+detail that fully communicates the proposal to your reviewers. If you're
+unsure, reach out to your manager for help.
+
+Remember to document any dependencies that may need to break or change as a
+result of this work.
+-->
+
+3. [Console UX and Workflows](#console-ux-and-workflows) sketches the main UX
+   workflows that the current design aims to support.
+1. [Catalog Schema](#catalog-schema) section describes the design of a catalog
+   schema that contains the optimizer notices corresponding to the current
+   catalog state.
+2. [Schema Maintenance](#schema-maintenance) elaborates how the contents of the
```

Comment:
"Schema Maintenance" sounds to me as if it were about changes to the schema itself, such as adding new columns.

Reply:
Renamed this to "Notice Generation and Management".